portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,19 +33,41 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private List<String> messages;
-
-  @Override
-  public void init() {
-    messages = new ArrayList<>(Arrays.asList("Hi there!", "Welcome to my protfolio", "Have a great day :)"));
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Send the JSON as the response
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String comment = (String) entity.getProperty("comment");
+      comments.add(comment);
+    }
+
     response.setContentType("application/json;");
-    response.getWriter().println(new Gson().toJson(messages)),
+    Gson gson = new Gson();
+    response.getWriter().println(gson.toJson(comments));
   }
 
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form
+    String comment = request.getParameter("comment");
```

Comment:
You can skip the spaces (same for the other method). In general we do not separate code lines by spacing, just keep reasonable order logic:
// Parameters definition
// Core logic
// Return

Reply:
https://google.github.io/styleguide/javaguide.html#s4.6-whitespace
It says in the java style guide that a single blank line may appear anywhere it improves readability, for example between statements to organize the code into logical subsections.  
Here I agree the spacing is not needed so I fixed it.

Reply:
Awesome:)